### DISCUSSION TITLE
Org mode plugin

### DISCUSSION THREAD

**Discussion by txtyash** [helix-editor/helix#10359] [General]
Org mode in helix would be great for note taking. If not built-in then as a separate plugin would also do the job.

**Comment by Niedzwiedzw** [comment]
yessssss!!!!!

**Comment by kodesoul** [comment]
If this feature is ever implemented could Neorg be considered in place of Org mode.

**Comment by txtyash** [comment]
@codesaf1 I don't think org mode would take a very long time. Devs seem to be very busy with a lot of other stuff. 

Yes, neorg is nice too.

**Comment by ubitux** [comment]
Using helix as a notes taking app / organizer is currently my main blocker for moving away from vim/neovim where there is the [Vimwiki](https://github.com/vimwiki/vimwiki) plugin. It's not exactly an equivalent to org-mode, but it can be thought as a markdown editor on steroid.

It might be possible to avoid the need for a plugin entirely by relying on a [markdown language server](https://github.com/microsoft/vscode-markdown-languageservice) but it's unclear to me yet if that would be enough.

The bare features I would be expecting from such a system would just be:
- pressing `<enter>` transforms the word under the cursor (or the current selection) into a link to a page of that word/selection (and pressing again jumps to that file):
- creating nested lists of checkable entries, with appropriate bindings to control their progress
- helper to create tables

I think this would cover about 90% of my notes usage. The rest is more specific to notes taking app (typically diary files management) while I believe these features remain in the editing realm.

Here is a demo of vimwiki with these 3 features:

https://user-images.githubusercontent.com/34467/201099154-ece4c011-1f07-436e-a1c7-775a60aaa612.mp4

I believe these are not so far-fetched for a general purpose markdown file support.

**Comment by edrex** [comment]
> Org mode in helix

This isn't very actionable. Org mode has been under continuous development for [20 years](https://orgmode.org/manual/History-and-Acknowledgments.html) and has a vast feature set built using Emacs Lisp. Let's focus on specific functionality we'd like to see in Helix.

- Support is already present (https://github.com/helix-editor/helix/pull/1845) for highlighting, although [the queries](https://github.com/helix-editor/helix/blob/master/runtime/queries/org/highlights.scm) could be expanded, eg to support ["todo keywords"](https://orgmode.org/worg/org-syntax.html#Headings). Not all themes look good with org files. `everforest` `_dark` and `_light` look nice.
- As @ubitux noted, a language server can be a nice way to add language-specific functionality, but org mode currently lacks any LSP server. See [ORG-Mode LSP : r/emacs](https://www.reddit.com/r/emacs/comments/et99uk/orgmode_lsp/), [tecosaur/org-lsp](https://github.com/tecosaur/org-lsp) (just an idea at present, :wave: @tecosaur), [Emacs as an Org LSP server : list.orgmode.org](https://list.orgmode.org/87zh2g2pby.fsf@bzg.fr/t/).
- Features not useful across languages are unlikely to be included in core. The path to getting these features will be either to figure out how to generalize them nicely across languages, or get the needed foundational enhancements into core to enable them. Examples: #3806, https://github.com/helix-editor/helix/issues/1840#issuecomment-147035330, #5737, #3134. https://github.com/helix-editor/helix/pull/5420 was a huge step forward, and more foundational stuff is in the pipeline.



> neorg

This issue is specifically about improving Org mode language support in Helix. Support for other languages could be added independently.

> markdown wiki

@ubitux also OT, but I'm with you. Let's continue Markdown wiki usability discussion [elsewhere](https://github.com/helix-editor/helix/discussions/1196#discussioncomment-5929742).

**Comment by AshtonKem** [comment]
I feel like the better approach here would not be to build in org-mode support into Helix, but instead create an org-mode LSP server that Helix could integrate with.

**Reply by Raxmo** [reply]
This is likely the way to go about it. Not only that, but it would also be the most portable and self-contained as well. I'm not too knowledgeable about LSP's specifically, but wouldn't said LSP be able to be used by other editors to give them org functionality as well? or would they need to also support the LSP in particular?
